### PR TITLE
Move more code out of the UI into "rules" and add basic test infra

### PR DESCRIPTION
**Pull request by robertos**

### CODE REVIEW

File: client/jest.config.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  preset: 'ts-jest',
+  testEnvironment: 'node',
+};
```
Comment (Zamiell) [inline client/jest.config.js:4]: all files should have newlines at the end to conform to unix/linux standards

**Comment by robertos** [comment]
@Zamiell to run tests use "npm test". I'm sending a new version that does that in the travis CI yml

File: client/src/game/rules/clues.ts [hunk 2]
```diff
@@ -1,19 +1,9 @@
+/* eslint-disable import/prefer-default-export */
```
Comment (Zamiell) [inline client/src/game/rules/clues.ts:1]: is this because there is only 1 exported function now?

Comment (robertos) [reply client/src/game/rules/clues.ts:1]: Yes, for now. It's a bit annoying.

File: client/src/game/rules/hand.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable import/prefer-default-export */
+// Functions related to hand management: card counts
+
+export function cardsPerHand(playerCount: number) {
+  switch (playerCount) {
+    case 2: case 3:
```
Comment (Zamiell) [inline client/src/game/rules/hand.ts:6]: does not conform to airbnb, see 15.5

**Comment by robertos** [comment]
https://travis-ci.com/github/Zamiell/hanabi-live/jobs/347827531

```
 PASS  src/game/rules/clues.test.ts (5.53 s)
 PASS  src/game/rules/deck.test.ts
Test Suites: 2 passed, 2 total
Tests:       11 passed, 11 total
Snapshots:   0 total
Time:        5.803 s
Ran all test suites.
The command "npm test" exited with 0.
```

**Comment by Zamiell** [comment]
thank you!!

File: client/src/game/rules/hand.ts (hunk 3 above)
Comment (robertos) [reply client/src/game/rules/hand.ts]: Ugh, ok

**Pull request merged**